src/lexer/lexer.cc

```diff
@@ -20,11 +20,31 @@ std::unique_ptr<std::istream> get_stream_from_source(const std::string& source,
 
 }  // namespace
 
-Lexer::Lexer(const std::string& source, SourceTag tag)
-  : stream_{get_stream_from_source(source, tag)} {}
+Lexer::Lexer(const std::string& source, SourceTag tag,
+             const std::string& filename)
+  : stream_{get_stream_from_source(source, tag)},
+    location_{filename, 1, 0} {}
 
-Token Lexer::get_next_token() {
-  return {TokenType::END_OF_FILE, "", location_, location_};
+
+ErrorOr<Token, LexError> Lexer::get_next_token() {
+  if (location_.line == 0)
+    return LexError("Invalid location", location_);
+  return {{TokenType::END_OF_FILE, "", location_, location_}};
+}
+
+MaybeError<LexError> Lexer::get_next_char() {
+    if (next_char_ == '\n') {
+        ++location_.line;
+        location_.column = 0;
+    }
+    ++location_.column;
+    if (!stream_->get(next_char_))
+        return makeError("Could not read from stream while parsing at position ");
```

Comment:
You don't need the "at position" since it's already added in the constructor of LexerError.